src/client/components/Analytics/index.js

```diff
@@ -0,0 +1,129 @@
+                : 0
+            }
+            valueMax={allTitles.length}
+            shouldAnimateOnMount
+            formatScreenReaderValue={({ valueNow, valueMax }) =>
+              `${valueNow} out of ${valueMax}`
+            }
+            renderValue={({ valueNow, valueMax }) => (
+              <span>
+                <Text size="medium" weight="bold">
+                  {valueNow}
+                </Text>
+                <br />
+                <Text size="small">/&nbsp;</Text>
+                <Text size="small">{valueMax}</Text>
+              </span>
+            )}
+          />
+        </View>
+      }
+    >
+      <Table caption="Analytics" hover>
+        <Table.Head>
+          <Table.Row>
+            <Table.ColHeader id="title">Media title</Table.ColHeader>
+            <Table.ColHeader id="progress">Progress</Table.ColHeader>
+          </Table.Row>
+        </Table.Head>
+        <Table.Body>
+          {allTitles.map(title => (
+            <Table.Row key={`title ${title}`}>
+              <Table.RowHeader>{title}</Table.RowHeader>
+              <Table.Cell>
+                <ProgressCircle
+                  size="small"
+                  screenReaderLabel="Percent complete"
+                  shouldAnimateOnMount
+                  formatScreenReaderValue={({ valueNow, valueMax }) =>
+                    `${Math.round((valueNow / valueMax) * 100)} percent`
+                  }
+                  renderValue={({ valueNow, valueMax }) => {
+                    const percent = Math.round((valueNow / valueMax) * 100);
+                    return (
+                      <span>
+                        <Text size="medium" weight="bold" color="primary">
+                          {percent > 100 ? 100 : percent}
+                        </Text>
+                        <Text size="xx-small" weight="bold" color="secondary">
+                          %
+                        </Text>
+                      </span>
+                    );
+                  }}
+                  valueMax={
+                    analytics.has(user.userid) &&
+                    analytics.get(user.userid).analytics.has(title)
+                      ? analytics.get(user.userid).analytics.get(title).time +
+                        analytics.get(user.userid).analytics.get(title)
+                          .remaining
+                      : 100
+                  }
+                  valueNow={
+                    analytics.has(user.userid) &&
+                    analytics.get(user.userid).analytics.has(title)
+                      ? analytics.get(user.userid).analytics.get(title)
+                          .finishedTimes
+                        ? analytics.get(user.userid).analytics.get(title).time +
+                          analytics.get(user.userid).analytics.get(title)
+                            .remaining
+                        : analytics.get(user.userid).analytics.get(title).time
+                      : 0
```

Comment:
@maxpluez It seems like for `analytics.get(user.userid).analytics.get(title).finishedTimes`, either way it will use `analytics.get(user.userid).analytics.get(title).time`, so maybe the code could be

```js
analytics.has(user.userid) &&
analytics.get(user.userid).analytics.has(title)
  ? analytics.get(user.userid).analytics.get(title).time +
    (analytics.get(user.userid).analytics.get(title).finishedTimes ?
      analytics.get(user.userid).analytics.get(title).remaining
        : 0)
  : 0
```

Though the code still seems a bit hard to read, wondering if there's a better way

Reply:
Yeah since I can't use if statement or variable assignments in JSX, I don't think there is any better way other than nested ternary without adding a lot of unnecessary data.